lib/twilio/device.ts

```diff
@@ -939,7 +939,15 @@ class Device extends EventEmitter {
 
     const config: Call.Config = {
       audioHelper: this._audio,
-      getUserMedia: this._options.getUserMedia || getUserMedia,
+      getUserMedia: (...args) =>
+        (this._options.getUserMedia || getUserMedia)(...args)
+        .then((gumResponse: any) => {
+          this._audio?._updateAvailableDevices().catch(error => {
+            // Ignore error, we don't want to break the call flow
+            this._log.warn('Unable to updateAvailableDevices after gUM call', error);
+          });
+          return Promise.resolve(gumResponse);
+        }),
```

Comment:
It seems like we're attaching extra responsibility to `gUM` here that it normally is not responsible for. Seems kinda strange, my gut instinct would be to pass the `_audio` to the Call so the Call can perform that logic itself after using `gUM`. Otherwise, this logic is kind of obscure and the Call `gUM` is intrinsically different than other places we use `gUM`.

Reply:
I did consider it. However, the Call object does not know anything about the audio helper. The Device object has that responsibility. Putting audio helper inside the Call object means, we're now sharing that between Call and Device, which is not necessary.

We're not attaching extra responsibility to gUM. The Device is responsible for audio devices, including the sequence of MediaDevice's API calls (gUM, enumerateDevices). This implementation is doing that.

Also, no need to call _updateAvailableDevices in other places where we call gUM. We only need it at the start of the call.

Reply:
Could we rename it from `gUM` to something else? I know it looks cumbersome but I don't like having unknown side-effects.

Reply:
I think that might create more confusion. Instead, I changed the implementation to use a callback instead. Please take another look.